src/concrete/ml/sklearn/tree_to_numpy.py

```diff
@@ -512,3 +524,227 @@ def get_lsbs_to_remove_for_trees(array: numpy.ndarray) -> int:
+    quantized_thresholds_array = numpy.empty(bias_1.shape, dtype=numpy.int64)
+    dequantized_thresholds_array = numpy.empty(bias_1.shape, dtype=numpy.float64)
+
+    for threshold_index, threshold_value in enumerate(bias_1[:, 0]):
+        feature_index = int(weight_1[threshold_index, :].argmax())
+        quantized_threshold_value = (
+            input_quantizers[feature_index].quant(threshold_value).astype(numpy.int64)
+        )
+        dequantized_threshold_value = input_quantizers[feature_index].dequant(
+            quantized_threshold_value
+        )
+        quantized_thresholds_array[threshold_index, 0] = quantized_threshold_value
+        dequantized_thresholds_array[threshold_index, 0] = dequantized_threshold_value
+
+    onnx_model.graph.initializer[bias_1_index].CopyFrom(
+        numpy_helper.from_array(
+            quantized_thresholds_array,
+            bias_1_name,
+        )
+    )
+
+    # Modify the graph inplace to keep only the parts that are of interest to us
+    tree_onnx_graph_preprocessing(onnx_model, framework, expected_number_of_outputs)
+
+    # Get the preprocessed tree predictions to replace the current
+    # (non-quantized) values in the onnx_model.
+    q_y = None
+    for initializer_index, initializer in enumerate(onnx_model.graph.initializer):
+        init_tensor = numpy_helper.to_array(initializer)
+        if "weight_3" in initializer.name:
+            # weight_3 is the prediction tensor
+            # Here we quantize it
+            q_y = preprocess_tree_predictions(init_tensor, n_bits)
+            init_tensor_as_int = q_y.qvalues.astype(numpy.int64)
+        else:
+            init_tensor_as_int = init_tensor.astype(numpy.int64)
+        assert (
+            isinstance(init_tensor_as_int, numpy.ndarray)
+            and init_tensor_as_int.dtype == numpy.int64
+        )
+        new_initializer = numpy_helper.from_array(init_tensor_as_int, initializer.name)
+        onnx_model.graph.initializer[initializer_index].CopyFrom(new_initializer)
+
+    # Convert the tree inference with Numpy operators
+    enable_rounding = bool(int(os.environ.get("TREES_USE_ROUNDING", 1)))
+
+    if not enable_rounding:
+        warnings.simplefilter("always")
+        warnings.warn(
+            "Using Concrete tree-based models without the `rounding feature` is deprecated. "
+            "Consider setting 'use_rounding' to `True` for making the FHE inference faster "
+            "and key generation.",
+            category=DeprecationWarning,
+            stacklevel=2,
+        )
+
+    lsbs_to_remove_for_trees: Optional[Tuple[int, int]] = None
+
+    assert q_y is not None
+    output_quantizers = [q_y.quantizer]
+
+    if enable_rounding:
+        # Quantize data to compute lsbs to remove
+        if model_inputs is None:
+            # If we have no data we can just randomly generate a dataset
+            assert isinstance(n_features, int)
+            calibration_set_size = 100_000
```

Comment:
do you really need 100 000 sampels here ?

Reply:
Probably not, it was to be safe when developing, I can reduce this

Reply:
Changed it to 1000